pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -99,7 +94,13 @@ func NewInstanceSettings(cfg *setting.Cfg, clientProvider *httpclient.Provider,
 			return nil, fmt.Errorf("error reading settings: %w", err)
 		}
 
-		cloud, err := getAzureCloud(cfg, jsonData)
+		var jsonData types.AzureClientSettings
```

Comment:
Could I suggest rather than doing multiple unmarshals, we create a `struct` composed of both `AzureClientSettings` and `AzureMonitorSettings` and then we will only have to unmarshal twice rather than 3 times 😊

Reply:
Thanks! Will fix it